Display SonarDelphi errors instead of failing scan

fourls opened the pull request:
This PR refactors the handling of errors that occur within SonarDelphi during analysis. Previously, any error-level logging would cause the analysis to fail with a fatal error, which is inaccurate - SonarDelphi is quite resilient to errors and can conduct a successful partial analysis in most cases an error appears. This new approach communicates the logs directly to the client _along_ with any results, leaving the user free to determine how important the error is / continue using DelphiLint while they identify the problem.

A new "View Last Analysis Log" option is available in the "Analyze" dropdown:

![image](https://github.com/user-attachments/assets/93fcc2a7-976c-454a-90d5-b55f39ff9d56)

Clicking this after an analysis shows a log window with options to save a log file or report an issue on integrated-application-development/sonar-delphi:

![image](https://github.com/user-attachments/assets/8b118c33-8727-44d7-859c-9af92afe47b3)

If the analysis produces errors, then the following shows up (clicking opens the log):

![image](https://github.com/user-attachments/assets/c094ae30-8dfe-4e36-bf44-cbbaf8102f58)

If the analysis produces warnings, then the following shows up (clicking opens the log):

![image](https://github.com/user-attachments/assets/4560cdd9-f8bd-483a-b801-7a86304dabdb)

Example of a log with errors:

![image](https://github.com/user-attachments/assets/0fd93def-deb2-4140-beed-06fcd4f09872)

Cirras commented on client/source/DelphiLint.LogViewer.dfm:
```diff
@@ -0,0 +1,272 @@
+      Width = 718
+      Height = 15
+      Caption = 
+        'Please note that errors or warnings during an analysis can affect ' +
+        'the quality of the results, including false positive or false negative ' +
+        'issues.'
```
Is this a bit verbose? I'd have probably written it in a much more terse fashion like below, though I'm conscious that you want a consistent tone in the DelphiLint hints/messages.
```
Errors/warnings may affect analysis quality, causing missing issues or false positives.
```

Cirras commented on client/source/DelphiLint.LogViewer.dfm:
```diff
@@ -0,0 +1,272 @@
+      Width = 592
+      Height = 15
+      Caption = 
+        'Errors are usually caused by attempting to analyze an invalid file ' +
+        'or by misconfiguring a sonar-project.properties.'
```
Similar to above.
```
Errors are typically caused by malformed source files, or a misconfigured project.
```

Cirras commented on client/source/DelphiLint.ToolFrame.dfm:
```diff
@@ -247,11 +247,75 @@ object LintToolFrame: TLintToolFrame
+        Width = 132
+        Height = 27
+        Align = alClient
+        Caption = '2 errors'
```
Why is the caption "2 errors" by default?

fourls replied:
It's just something to look at at design-time. Do you think I should make it something else?

Cirras replied:
Nah, that seems fine.

Cirras commented on client/source/DelphiLint.ToolFrame.dfm:
```diff
@@ -264,10 +328,292 @@ object LintToolFrame: TLintToolFrame
+    end
+    object ViewLogItem: TMenuItem
+      Caption = '&View Last Analysis Log'
+      OnClick = ErrorButtonClick
```
This event handler name references the error button but is also used by the `WarningButton` and `ViewLogItem`. Should the name be something else?

fourls replied:
Changed to `ViewLogClick`.

Cirras reviewed (changes requested):
Great change.

Surfacing errors and warnings in this way is a nice win for consistency with standard SonarDelphi scans, and much less disruptive when things do go wrong.
I also really like the Analysis Log UI and hope the `Report a bug on GitHub` button will drive some useful bug reports to the upstream SonarDelphi project.

Testing went well.
- Should we have more visual feedback on the `Errors` and `Warnings` buttons?
  - The little magnifying glass is cute, but easy to miss. 
  - Tooltip on hover saying "View analysis log" or similar
  - Background color change on hover, like with other buttons.
- The GitHub icon in the Analysis Log is a nice touch, but there's some white pixels around it that look a little funny. BMP antialiasing issue?
  ![image](https://github.com/user-attachments/assets/21078a0d-9782-46b1-a49a-bae338ad9fbe)



Code changes are mostly good to go with a few nitpicks, see below.

fourls commented:
(rebase)

fourls commented:
Updated now based on feedback.

Log description:

![image](https://github.com/user-attachments/assets/f3d16ee5-3ad1-4dd9-8c9c-3da775567db6)

GitHub icon (dark theme):

![image](https://github.com/user-attachments/assets/6a43409a-a8a8-42cd-85a2-c574e8070fd7)

GitHub icon (light theme):

![image](https://github.com/user-attachments/assets/663e8969-9e80-435d-8358-9d922d725c5b)

> Should we have more visual feedback on the Errors and Warnings buttons?
> * The little magnifying glass is cute, but easy to miss.
> * Tooltip on hover saying "View analysis log" or similar
> * Background color change on hover, like with other buttons.

VCL is surprisingly restrictive with what it lets you do with buttons. I've added a tooltip, but the background color is totally dependent on the theme. In light mode, for example, it actually does darken the buttons:

![light_mode_errors](https://github.com/user-attachments/assets/6a3e5fc7-7dd5-4684-8041-03c58bb2c8ac)

There's not too much I can do without implementing a custom control. I don't think what we have now will be too much of a problem - it's so colourful and angry that I expect most people will click on it anyway.

Cirras reviewed (approved):
Looks good to me. 🎉
